File: src/view/SpinnerView.ts
import { bridge } from '../native/bridge'
import { native } from '../native/native'
import { View } from './View'

@bridge('dezel.view.SpinnerView')

/**
 * @class SpinnerView
 * @super View
 * @since 0.1.0
 */
export class SpinnerView extends View {

	//--------------------------------------------------------------------------
	// Properties
	//--------------------------------------------------------------------------

	/**
	 * The spinner view's tint color.
	 * @property tint
	 * @since 0.1.0
	 */
	@native public tint!: string

	/**
	 * Whether the spinner view is spinning.
	 * @property spin
	 * @since 0.1.0
	 */
	@native public spin!: boolean

}

File: platforms/ios/Dezel/Dezel/View/SpinnerView.swift
/**
 * @class SpinnerView
 * @super UIActivityIndicatorView
 * @since 0.7.0
 */
open class SpinnerView: UIActivityIndicatorView {

	//----------------------------------------------------------------------
	// MARK: Properties
	//----------------------------------------------------------------------

	/**
	 * @property spin
	 * @since 0.7.0
	 */
	open var spin: Bool = false {
		willSet {
			if (newValue) {
				self.startAnimating()
			} else {
				self.stopAnimating()
			}
		}
	}

	//--------------------------------------------------------------------------
	// MARK: Methods
	//--------------------------------------------------------------------------

	/**
	 * @constructor
	 * @since 0.7.0
	 */
	required public init(coder: NSCoder) {
		fatalError("init(coder:) has not been implemented")
	}

	/**
	 * @constructor
	 * @since 0.7.0
	 */
	public override init(frame: CGRect) {
		super.init(frame: frame)
		self.style = .gray
		self.hidesWhenStopped = false
	}
}
